pkg/interceptors/cel/cel.go

```diff
@@ -57,52 +60,136 @@ var (
 	mapType    = reflect.TypeOf(&structpb.Struct{})
 )
 
+type params = triggersv1.CELInterceptor
+
 // NewInterceptor creates a prepopulated Interceptor.
-func NewInterceptor(cel *triggersv1.CELInterceptor, k kubernetes.Interface, ns string, l *zap.SugaredLogger) interceptors.Interceptor {
+func NewInterceptor(k kubernetes.Interface, l *zap.SugaredLogger) *Interceptor {
 	return &Interceptor{
-		Logger:                 l,
-		CEL:                    cel,
-		KubeClientSet:          k,
-		EventListenerNamespace: ns,
+		Logger:        l,
+		KubeClientSet: k,
 	}
 }
 
-// ExecuteTrigger is an implementation of the Interceptor interface.
-func (w *Interceptor) ExecuteTrigger(request *http.Request) (*http.Response, error) {
-	env, err := makeCelEnv(request, w.EventListenerNamespace, w.KubeClientSet)
+// We need to keep this method around till we migrate the other interceptors to the new interface
+func (w *Interceptor) ExecuteTrigger(_ *http.Request) (*http.Response, error) {
+	return nil, fmt.Errorf("executeTrigger() is deprecated. Call Process() instead")
+}
+
+func evaluate(expr string, env *cel.Env, data map[string]interface{}) (ref.Val, error) {
+	parsed, issues := env.Parse(expr)
+	if issues != nil && issues.Err() != nil {
+		return nil, fmt.Errorf("failed to parse expression %#v: %s", expr, issues.Err())
```

Comment:
Return a wrapped error (same with other occurrences).
```suggestion
		return nil, fmt.Errorf("failed to parse expression %#v: %w", expr, issues.Err())
```

Reply:
done.